app/controllers/blog_controller.rb

```diff
@@ -107,6 +93,31 @@ def send_blog_file(relative_path, **kwargs)
     render_blog_404
   end
 
+  def blog_html_with_additions(absolute_path)
+    html = File.read(absolute_path)
+
+    html.sub!(
+      '</head>',
+      "#{helpers.csrf_meta_tags}#{helpers.window_data_script_tag}</head>",
+    )
+
+    html.sub!(
+      '</body>',
+      "#{helpers.ts_tag('comments')}</body>",
+    )
+
+    if params[:action] == 'show'
+      html.sub!(
+        '</head>',
+        "#{helpers.ts_tag('styles')}</head>",
+      )
+    end
```

Comment:
Here's the new part; the rest of this method is essentially just extracted from a different location (with a slight tweak to use `sub!` rather than `sub`).